multiwallet.go

```diff
@@ -598,23 +606,25 @@ func (mw *MultiWallet) ChangePrivatePassphraseForWallet(walletID int, oldPrivate
 		return errors.New(ErrInvalid)
 	}
 
-	err := wallet.changePrivatePassphrase(oldPrivatePassphrase, newPrivatePassphrase)
-	if err != nil {
-		return translateError(err)
-	}
+	if wallet.EncryptedSeed != nil {
+		decryptedSeed, err := decryptWalletSeed(oldPrivatePassphrase, wallet.EncryptedSeed)
+		if err != nil {
+			return err
+		}
 
-	decryptedSeed, err := decryptWalletSeed(newPrivatePassphrase, wallet.EncryptedSeed)
-	if err != nil {
-		return err
+		encrytedSeed, err := encryptWalletSeed(newPrivatePassphrase, decryptedSeed)
+		if err != nil {
+			return err
+		}
+
+		wallet.EncryptedSeed = encrytedSeed
 	}
 
-	encrytedSeed, err := encryptWalletSeed(newPrivatePassphrase, decryptedSeed)
+	err := wallet.changePrivatePassphrase(oldPrivatePassphrase, newPrivatePassphrase)
 	if err != nil {
-		return err
+		return translateError(err)
 	}
 
-	wallet.EncryptedSeed = encrytedSeed
-
 	wallet.PrivatePassphraseType = privatePassphraseType
 	return mw.db.Save(wallet)
```

Comment:
Should do more than return this error to the caller. Also undo the wallet passphrase change if saving the new encrypted seed to db fails; otherwise the user would not be able to decrypt this wallet's seed subsequently using the new password.

If trying to change back the password errors, we should return an error message that helps the caller understand that there's gonna be some discrepancies between the wallet private passphrase and the seed encryption passphrase.

Something like:
```suggestion
	err := mw.db.Save(wallet)
	if err != nil && wallet.EncryptedSeed != nil {
		log.Errorf("error saving re-encrpyted wallet seed: %v", err)
		err2 := wallet.changePrivatePassphrase(newPrivatePassphrase, oldPrivatePassphrase)
		if err2 != nil {
			log.Errorf("error undoing wallet passphrase change: %v", err2)
			return errors.New("wallet passphrase changed but re-encrypting seed failed")
		}
	}
	return err
```

Reply:
Check this out 4f5e49e, It holds the newly encrypted seed in a variable and sets it after changing the wallet private passphrase.

Reply:
It's quite the same thing. In memory, the encrypted seed uses the new password. But if the save fails, the db still retains the seed encrypted with the old password.